deltachat-ffi/src/lib.rs

```diff
@@ -1447,7 +1470,9 @@ pub unsafe extern "C" fn dc_block_chat(context: *mut dc_context_t, chat_id: u32)
         ChatId::new(chat_id)
             .block(ctx)
             .await
-            .ok_or_log_msg(ctx, "Failed chat block");
+            .context("Failed chat block")
+            .log_err(ctx)
+            .ok();
```

Comment:
I must say, I liked the old style better. If rustfmt allowed to format the new code as:
```rust
        ChatId::new(chat_id)
            .block(ctx).await
            .context("Failed chat block").log_err(ctx).ok();
```
I'd like the proposal. Alas, it doesn't (and I don't think we should stop using rustfmt), and I find it hard to visually tell apart the error handling from the rest of the code, and the error handling part just gets unnecessary "weight", making it harder to spot the parts where we have our actual logic.

BTW, we already had a very similar discussion, with similar arguments, in https://github.com/deltachat/deltachat-core-rust/pull/2284 (doesn't mean we can't re-open the discussion, of course):

> consider this code:
> 
> ```rust
>     if let Err(e) = context
>         .set_config(Config::LastHousekeeping, Some(&time().to_string()))
>         .await
>     {
>         warn!(context, "Can't set config: {}", e);
>     }
> ```
> 
> If we replace this with:
> 
> ```rust
>     context
>         .set_config(Config::LastHousekeeping, Some(&time().to_string()))
>         .await
>         .context("Can't set config");
>         .log(context)
>         .ok();
> ```
> 
> this doesn't seem to be that much of an improvement, while:
> 
> ```rust
>     context
>         .set_config(Config::LastHousekeeping, Some(&time().to_string()))
>         .await
>         .ok_or_log_msg(context, "Can't set config");
> ```
> 
> looks better to me.

Reply:
> I must say, I liked the old style better

I was afraid you would... given you argued to get it in the first place.

> I must say, I liked the old style better. If rustfmt allowed to format the new code as:
> 
> ```rust
>         ChatId::new(chat_id)
>             .block(ctx).await
>             .context("Failed chat block").log_err(ctx).ok();
> ```

This to me is kind of a weird argument: code formatters are there so I don't have to worry about formatting.  Moving the formatting considerations to the API design is somewhat counter productive.  I honestly don't care about what the formatter does, I care about a good API and a composable API here has benefits: it's flexible, expressive and, err, composable.


> BTW, we already had a very similar discussion, with similar arguments, in #2284 (doesn't mean we can't re-open the discussion, of course):

Well, clearly I've had a long time to consider this and am still of the same opinion that the composable API is better.  I mean, this was inspired by a change that was painful because it wasn't composable.

Reading that issue I'm shocked to find that I proposed the `.ok_or_log()` naming, I guess I was trying to compromise but I  consider that weird in relation to the standard `.ok_or()` and `.ok_or_else()` APIs.  It just gets the concepts wrong.  But then it's because it's trying to do two things that should be composed.